### PR TITLE
Add wrapper SQL types

### PR DESCRIPTION
**Pull request by erizocosmico**
Only last commit is from this PR

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	bytes, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	return bytes, nil
+}
+
+// SQLType is the common interface a type has to fulfill to be considered a
+// SQL type.
+type SQLType interface {
```
Comment (erizocosmico) [inline types/types.go]: If find `SQLType` way more explicit than `ScanValuer`

File: types/types.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	return fmt.Errorf("cannot scan type %s into JSON type", reflect.TypeOf(v))
+}
+
+// JSONValue converts something into json.
+// WARNING: This is here temporarily, might be removed in the future, use
+// `JSON` instead.
+func JSONValue(v interface{}) (driver.Value, error) {
+	bytes, err := json.Marshal(v)
```
Comment (erizocosmico) [inline types/types.go]: right

**Comment by erizocosmico** [comment]
Done @dpordomingo

File: types/types.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	case string:
+		url, err := url.Parse(t)
+		if err != nil {
+			return fmt.Errorf("error scanning url: %s", err)
+		}
+
+		*u = URL(*url)
+		return nil
+	}
+	return fmt.Errorf("cannot scan type %s into URL type", reflect.TypeOf(v))
```
Comment (Serabe) [inline types/types.go]: What is the difference between `reflect.TypeOf` and using `%T`?

**Pull request merged**